nodejs/eks/nodegroup.ts

```diff
@@ -652,19 +653,54 @@ export function createNodeGroup(
     return createNodeGroupInternal(name, args, pulumi.output(core), parent, provider);
 }
 
+export function resolveOrGetInstanceProfile(
```

Comment:
Could you please add function docs explaining what this function is exactly doing and when it's throwing errors. Thanks!